src/operator/tensor/matrix_op-inl.h

```diff
@@ -560,7 +561,9 @@ void SliceCsrImpl(const SliceParam &param, const OpContext& ctx,
   CHECK_NE(req, kAddTo) << "kAddTo for Slice on CSR input is not supported";
   CHECK_NE(req, kWriteInplace) << "kWriteInplace for Slice on CSR input is not supported";
   int begin = *param.begin[0];
+  if (begin < 0) begin += ishape[0];
   int end = *param.end[0];
+  if (end < 0) end += ishape[0];
```

Comment:
Could you make sure we have the same behavior as scipy?
if shape[0] = 5, we should disallow nd[-100] (correct me if scipy is not like this)

Reply:
If `begin` or `end` is invalid, checks in infer_shape https://github.com/apache/incubator-mxnet/blob/master/src/operator/tensor/matrix_op-inl.h#L382 will fail. For example, if `nd[-100]`, here is the err msg,
```
>>> mx.nd.slice(x, begin=-100, end=-99)
[09:28:50] /home/ubuntu/zyhmxnet/dmlc-core/include/dmlc/./logging.h:308: [09:28:50] src/operator/tensor/./matrix_op-inl.h:408: Check failed: s >= 0 && s < e && e <= static_cast<int>(dshape[i]) Invalid slicing begin (-100,) and end (-99,) for data of shape (2,2)
```